packages/compass-e2e-tests/helpers/commands/shell-eval.ts

```diff
@@ -3,11 +3,11 @@ import * as Selectors from '../selectors';
 
 async function getOutputText(browser: CompassBrowser): Promise<string[]> {
   const elements = await browser.$$(Selectors.ShellOutput);
-  return Promise.all(
-    elements.map((element) => {
-      return element.getText();
-    })
-  );
+  const promises = elements.map(async (element) => {
+    return await element.getText();
+  });
+  const result = await Promise.all(promises as unknown as Promise<string>[]); // TODO: something weird happening
```

Comment:
I don't know why typescript seems to be getting confused. Every time we use this `elements = await browser.$$(); await Promise.all(elements.map((element) => element.returnSomePromise()));` pattern

Reply:
Well, `map` seems to be returning a `Promise<string[]>`, not a `Promise<string>[]` (i.e. as if `Promise.all()` had already been applied) – is the webdriverio type here incorrect?

Reply:
Looks like it, yeah.

Reply:
ok I was being dumb. They have their own map method that's actually kinda nice.